tabular/tests/conftest.py

```diff
@@ -19,3 +26,118 @@ def pytest_collection_modifyitems(config, items):
+
+    @staticmethod
+    def load_data(directory_prefix, train_file, test_file, name, url=None):
+        if not os.path.exists(directory_prefix):
+            os.mkdir(directory_prefix)
+        directory = directory_prefix + name + "/"
+        train_file_path = directory + train_file
+        test_file_path = directory + test_file
+        if (not os.path.exists(train_file_path)) or (not os.path.exists(test_file_path)):
+            # fetch files from s3:
+            print("%s data not found locally, so fetching from %s" % (name, url))
+            zip_name = ag.download(url, directory_prefix)
+            ag.unzip(zip_name, directory_prefix)
+            os.remove(zip_name)
+
+        train_data = task.Dataset(file_path=train_file_path)
+        test_data = task.Dataset(file_path=test_file_path)
+        return train_data, test_data
+
+
+class FitHelper:
+    @staticmethod
+    def fit_and_validate_dataset(dataset_name, fit_args, sample_size=1000, refit_full=True, delete_directory=True):
+        directory_prefix = './datasets/'
+        train_data, test_data, dataset_info = DatasetLoaderHelper.load_dataset(name=dataset_name, directory_prefix=directory_prefix)
+        label_column = dataset_info['label_column']
+        directory = directory_prefix + dataset_name + "/"
+        savedir = directory + 'AutogluonOutput/'
+
+        shutil.rmtree(savedir, ignore_errors=True)  # Delete AutoGluon output directory to ensure runs' information has been removed.
+        fit_args['label'] = label_column
+        fit_args['output_directory'] = savedir
+        predictor = FitHelper.fit_dataset(train_data=train_data, fit_args=fit_args, sample_size=sample_size)
+        if sample_size is not None and sample_size < len(test_data):
+            test_data = test_data.sample(n=sample_size, random_state=0)
+        predictor.predict(test_data)
+        predictor.predict_proba(test_data)
+        model_names = predictor.get_model_names()
+        model_name = model_names[0]
+        assert len(model_names) == 2
+        if refit_full:
+            refit_model_names = predictor.refit_full()
+            assert len(refit_model_names) == 2
+            refit_model_name = refit_model_names[model_name]
+            assert '_FULL' in refit_model_name
+            predictor.predict(test_data, model=refit_model_name)
+            predictor.predict_proba(test_data, model=refit_model_name)
+        predictor.info()
+        predictor.leaderboard(test_data, extra_info=True)
+        assert savedir == predictor.output_directory
+        if delete_directory:
+            shutil.rmtree(savedir, ignore_errors=True)  # Delete AutoGluon output directory to ensure runs' information has been removed.
+        return predictor
+
+    @staticmethod
+    def fit_dataset(train_data, fit_args, sample_size=None):
+        if sample_size is not None and sample_size < len(train_data):
+            train_data = train_data.sample(n=sample_size, random_state=0)
```

Comment:
should this be stratifed by target variable to ensure cases when in multiclass we could actually test binary classification?

Reply:
stratification won't actually solve that since we still drop classes with low count if eval_metric allows it (aka accuracy). Regardless, I don't think we need to worry about this edge case, since I am using large enough samples (1000) that it won't occur. The way to ensure it works would be via targeting log_loss + stratified sampling, but we probably don't need to go that far by having reasonable sample sizes.